Zend/Optimizer/zend_inference.c

```diff
@@ -3682,9 +3682,7 @@ static zend_always_inline zend_result _zend_update_type_info(
 					UPDATE_SSA_TYPE(tmp, ssa_op->op1_def);
 				} else {
 					/* invalid key type */
-					tmp = (tmp & (MAY_BE_RC1|MAY_BE_RCN|MAY_BE_ARRAY)) |
-						(t1 & ~(MAY_BE_RC1|MAY_BE_RCN|MAY_BE_UNDEF|MAY_BE_NULL|MAY_BE_FALSE));
-					UPDATE_SSA_TYPE(tmp, ssa_op->op1_def);
+					return SUCCESS;
```

Comment:
I have a question out of interest. Why do you return SUCCESS here? Doesn't that leave the type inference for some opcodes not yet determined?

Reply:
Returning FAILURE will completely stop the type inference.

Here we have an instruction that throws exception and don't produce any results, however, future inference may process the same instruction again and that time it's going to be valid. See the test case...

Reply:
Ok got it, thank you.